main.cpp

```diff
@@ -1,46 +1,52 @@
+#include <cmath>
 #include <iostream>
-
+#define _USE_MATH_DEFINES
 using namespace std;
 
 /**
- * \brief Вычисляет площадь прямоугольника.
- * \param length Длина стороны прямоугольника.
- * \param width Ширина стороны прямоугольника.
- * \return Площадь прямоугольника.
- */
-double Square(const double length, const double width);
+* \brief Это функция выполняет вычисление функции при x < 1.2
+* \param a параметр а
+* \param x параметр х
+* \return значение функции
+*/
+double getY1(const double a, double x);
 
 /**
- * \brief Вычисляет периметр прямоугольника.
- * \param length Длина стороны прямоугольника.
- * \param width Ширина стороны прямоугольника.
- * \return Периметр прямоугольника.
- */
-double Perimeter(const double length, const double width);
+* \brief Это функция выполняет вычисление функции при x >= 1.2
+* \param a параметр а
+* \param x параметр х
+* \return значение функции
+*/
+double getY2(const double a, double x);
 
 /**
- * \brief Точка входа в программу.
- * \return Код ошибки (0 - успех).
- */
+* \brief Точка входа в программу
+* \return 0 в случае успеха
+*/
 int main()
 {
-    cout << "Введите длины сторон прямоугольника\n";
-    double a,b;
-    cin >> a >> b;
-
-    const auto square = Square(a, b);
-    cout << "Площадь квадрата равна " << square << "\n";
-
-    const auto perimeter = Perimeter(a, b);
-    cout << "Периметр прямоугольника " << perimeter << endl;
+double x;
+double y;
+const double a = 2;
+cout << "Введите значение х ";
+cin >> x;
+if (x > a)
+{
+y = getY1( a, x);
+cout << "y = " << y << "\n";
 }
-
-double Square(const double length, const double width)
+else
 {
-    return length * width;
+y = getY2( a, x);
+cout << "y = "<< y << "\n";
 }
-
-double Perimeter(const double length, const double width)
+return 0;
+}
+double getY1(const double a, double x)
+{
+return tan(x)+M_PI/3;
+}
+double getY2(const double a, double x)
```

Comment:
Если параметры функции не изменяются, то они явно должны быть обозначены `const`
[By Alex Allain](https://www.cprogramming.com/tutorial/const_correctness.html)
>The const keyword allows you to specify whether or not a variable is modifiable. You can use const to prevent modifications to variables and const pointers and const references prevent changing the data pointed to (or referenced).
>Const дает вам возможность более четко документировать вашу программу и фактически обеспечивать соблюдение этой документации. Ключевое слово const обеспечивает вашим пользователям гарантии, позволяющие оптимизировать производительность без угрозы повреждения их данных, обеспечивая соблюдение вашей документации. Например, константные ссылки позволяют указать, что упомянутые данные не будут изменены; это означает, что вы можете использовать константные ссылки как простой и немедленный способ повышения производительности для любой функции, которая в настоящее время принимает объекты по значению, не беспокоясь о том, что ваша функция может изменить данные. Даже если это произойдет, компилятор предотвратит компиляцию кода и предупредит вас о проблеме. С другой стороны, если бы вы не использовали константные ссылки, у вас не было бы простого способа гарантировать, что ваши данные не были изменены.